model/app.py

```diff
@@ -222,24 +262,28 @@ def key_from_model_info(wild_type_id, message):
     """
     d = {k: message.get(k, []) for k in {GENOTYPE_CHANGES, MEDIUM, MEASUREMENTS, REACTIONS_ADD, REACTIONS_KNOCKOUT}}
     d['model_id'] = wild_type_id
+
+    if version:
+        d['version'] = version
+
     return hashlib.sha224(json.dumps(d, sort_keys=True).encode('utf-8')).hexdigest()
 
 
-async def save_changes_to_db(model, wild_type_id, message):
+async def save_changes_to_db(model, wild_type_id, message, version=None):
     """Store model in cache database
 
     :param model: Cameo model
     :param wild_type_id: str
     :param message: dict
-    :return: str (cache database key)
+    :return: mutated_model_id (cache database key)
     """
-    db_key = key_from_model_info(wild_type_id, message)
+    mutated_model_id = key_from_model_info(wild_type_id, message, version)
     redis = await redis_client()
-    await redis.set(db_key,
+    await redis.set(mutated_model_id,
                     json.dumps({'model': model.id, 'changes': model.notes.get('changes', deepcopy(EMPTY_CHANGES))}))
```

Comment:
the same json patch can actually be used in the future to restore models and apply diffs without ever using model changes, and get rid of keeping the changes "manually" in Adapter class, but it is more work, so it's a great iterative step